lib/mix/lib/mix/tasks/usage.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Mix.Tasks.Usage do
+  use Mix.Task
+
+  @shortdoc "Prints usage information for mix"
+
+  @moduledoc ~S"""
+  Usage: mix [task]
+    mix                       - Runs the default task (current: "mix run")
+    mix new                   - Creates a new Elixir project
+
+  For further help:
+    mix help                  - Prints help information for tasks
+    mix help                  - Prints all tasks and their shortdoc
+    mix help TASK             - Prints full docs for the given task
+    mix help --search PATTERN - Prints all tasks that contain PATTERN in the name
+    mix help --names          - Prints all task names and aliases
+    mix cmd                   - Executes the given command
+    mix compile               - Compiles source files
+    mix deps                  - Lists dependencies and their status
+    mix do                    - Executes the tasks separated by comma
+    mix escript               - Lists installed escripts
+    mix loadconfig            - Loads and persists the given configuration
+    mix local                 - Lists local tasks
+    mix local.hex             - Installs Hex locally
+    mix local.public_keys     - Manages public keys
+    mix local.rebar           - Installs Rebar locally
+    mix profile.cprof         - Profiles the given file or expression with cprof
+    mix profile.fprof         - Profiles the given file or expression with fprof
+    mix run                   - Runs the given file or expression
+    iex -S mix                - Starts IEx and runs the default task
+  """
```

Comment:
Well. I think it will be a burden, and one that people will forget to update as the `@shortdoc`s change.
You don't need to be inside a Mix project to run `mix help`, so I don't see why you will need to Mix.Project file.
I think if you make Mix.Tasks.build_task_doc_list/1 public with `@doc false`, you can still pass the list of modules, and they will be returned in the same order. The one ordering the tasks is `load_tasks/0` and you don't need that one.
Sorry, I didn't run any test, these are all assumptions by quickly reading the code. I could be wrong, so please let me know